### ISSUE TITLE
[question] pkg_config generator with layout generates invalid .pc file

### ISSUE THREAD

**Issue by joastoe** [conan-io/conan#9897]
I'm quite new to conan and trying to integrate it into our current build flow (not from scratch).

We use cmake which finds its dependencies with pkg_config (pkg_search_module). So I set up a test of two libs one requiring another, LibA depends on LibB. So when building LibA the pkg_config generator succesfully generates the .pc files

```
prefix=/home/.conan/data/libB/1.0.0_conan/company/testing/package/abd3ca9581f5ec3d6672fa2ee8818b1f09dbb082
libdir=${prefix}/lib
includedir=${prefix}/include

Name: libB-lib
Description: Generic purpose library for CPP development
Version: 1.0.0_conan
Libs: -L"${libdir}" -lcomet  -lcomet-tests  -Wl,-rpath,"${libdir}"
Cflags: -I"${includedir}"
```

So far so good, now I wanted to switch to layouts and to use the editable feature.

I changed the conanfile of libB from 
```
  def build(self):
        cmake = CMake(self)
        cmake.verbose = True
        cmake.definitions["COMPILE_INTEGRATION_TESTS"] = "OFF"
        cmake.definitions["GENERATE_CODE_COVERAGE"] = "OFF"
        cmake.definitions["RUN_TESTS"] = "OFF"
        cmake.definitions["CMAKE_INSTALL_PREFIX"] = "deploy"
        cmake.configure(build_folder="build", source_folder=".")
        cmake.build()
        cmake.install()

    def package(self):
        self.copy("*", src="build/deploy")

    def package_info(self):
        self.cpp_info.libs = tools.collect_libs(self)
```

to

```
def layout(self):
        # ###### FOLDERS
        self.folders.build = "build"

        # We want to have the toolchains in the build folder so we can always pass
        # `-DCMAKE_TOOLCHAIN_FILE=generators/conan_toolchain.cmake` to CMake
        self.folders.generators = os.path.join(self.folders.build, "generators")

        # In case we use "conan package" we declare an output directory
        self.folders.package = "deploy"

        # ###### INFOS
        self.cpp.source.includedirs = ["src"] # Relative to ["."] (self.folders.source)
        self.cpp.build.libdirs = ["deploy/lib"]  # Relative to (self.folders.build)

    def build(self):
        cmake = CMake(self)
        cmake.verbose = True
        cmake.definitions["COMPILE_INTEGRATION_TESTS"] = "OFF"
        cmake.definitions["GENERATE_CODE_COVERAGE"] = "OFF"
        cmake.definitions["RUN_TESTS"] = "OFF"
        cmake.definitions["CMAKE_INSTALL_PREFIX"] = "deploy"
        cmake.configure()
        cmake.build()
        cmake.install()

    def package(self):
        LayoutPackager(self).package()

    #def package_info(self):
    #    self.cpp_info.libs = tools.collect_libs(self)
```

Now the pkg generator of LibA creates the following .pc file
```
prefix=/home/.conan/data/libB/1.0.0_conan/company/testing/package/abd3ca9581f5ec3d6672fa2ee8818b1f09dbb082

Name: libB-lib
Description: Conan package: libB
Version: None
Libs: 
Cflags: 
```

What am I doing wrong?
What makes it hard for me to track down the problem is, that I don't know where actually the information comes from to generate the .pc file. I can not find anything in the .conan folder.

- [x ] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @joastoe 

It seems you are mixing things. If you aim to use the ``layout()`` you should be using the modern build system integrations:
- PkgConfigDeps, and not ``pkg_config``
- CMake is from ``from conan.tools.cmake import CMake``, and should be used together with CMakeToolchain and CMakeDeps

You want to use things from this section https://docs.conan.io/en/latest/reference/conanfile/tools.html to use the ``layout()``.

The information that goes into generated files (.pc by ``PkgConfigDeps``) always come from the ``package_info()`` method of the dependencies, that must be complete and correct.

**Comment by joastoe** [comment]
Hi @memsharded

Thanks very much for your response. With your info provided, I was able solve my problems.

One thing that came to mind: Is it also possible to make advantage of the **editable** feature without using `layout()` and the modern build system integrations?

**Comment by memsharded** [comment]
> One thing that came to mind: Is it also possible to make advantage of the editable feature without using layout() and the modern build system integrations?

If you are thinking about the legacy "editable" layout definition, that used external template jinja files, no, it will not play well together. The new build system integrations already listen and follow the new layout definition, having the ``layout()`` method or not.

**Comment by joastoe** [comment]
Another issue regarding **pkg config** I'm fighting with:

I have libA->libB->libC

When building libA with the pkg_config generator I get the following generated .pc file for libB
```
prefix=/.conan/data/libB/1.0.0_test/company/testing/package/13ca7c3421ced37746801df275efb2bb7f8fca28
libdir=${prefix}/lib
includedir=${prefix}/include

Name: libB
Description: Library B
Version: 1.0.0_test
Libs: -L"${libdir}" -lds  -lds-tests  -Wl,-rpath,"${libdir}"
Cflags: -I"${includedir}"
Requires: libC
```

After switching to PkgConfigDeps, the generated .pc file changes:
```
prefix=/.conan/data/libB/1.0.0_test/company/testing/package/13ca7c3421ced37746801df275efb2bb7f8fca28
libdir=${prefix}/lib
includedir=${prefix}/include

Name: libB
Description: Library B
Version: 1.0.0_test
Libs: -L"${libdir}" -lds  -lds-tests  -Wl,-rpath,"${libdir}"
Cflags: -I"${includedir}"
```
The **Requires** attribute is missing. 
Can someone help me out with this?

Thanks very much

**Comment by memsharded** [comment]
@joastoe If the issue above was solved, then maybe we can close this issue, and open your last comment as a new one? Seems not the same issue.

Please also complete, just in case, are libA->libB->libC regular ``requires``? No ``build_requires`` or any other different type of requirement?

**Comment by joastoe** [comment]
Of course you're right @memsharded, although the issue is quite related.
I opened a new issue here: [Indirect dependencies missing when using PkgConfigDeps](https://github.com/conan-io/conan/issues/9939)
This issue can be closed, indeed. Thanks!
